### DISCUSSION TITLE
Recommended development boards

### DISCUSSION THREAD

**Discussion by minexn** [kernelkit/infix#289] [Q&A]
Are there any boards you use for development or recommend testing Infix?

**Comment by wkz** [comment]
For the time being, we are mostly focused on customer boards that are not publicly available. But the designs center around the CN9130 SoC connected to a 88E6393X switch (both from Marvell), and there is a reference board for this combo from Marvell: "Marvell CN9130 CRB".  We have run Infix on this board before, and it would probably be pretty easy to bring that back up again. Unfortunately, those boards seem hard to come by. 

Another board we've looked at, but not yet done any work with, is the [RB5009UB](https://mikrotik.com/product/rb5009ug_s_in) from MikroTik. This has the same switch, and a smaller version of the same SoC.

**Reply by troglobit** [reply]
The current focus is definitely on the Marvell SOHO/Linkstreet line of switchcores, we know them very well, so they've been a safe design choice for customer deliveries (single or multiple switchcores in a cascade for more ports).  The MikroTik board (above) is *very* close in functionality to one of our customers, and it's affordable, so very likely to become our reference board for Infix, I'd say.

Future projects are, however, likely to lean more towards Microchip.  Their [LAN966x (*Laguna*)](https://www.microchip.com/en-us/product/lan9694) family of switches, following on from the mighty SparX-5i (*Fireant*), fits much of our customer base like a glove.  (Don't let the number of ports scare you, it's the core switching performance that's the important metric.)  They also have the smaller switches in the [KSZ-line (*Maserati*)](https://www.microchip.com/en-us/product/KSZ9477), which we are also looking at for some projects.  Microchip are also very pro Open Source, even for their high end chipsets, and actively supporting their kernel drivers.

Other boards may appear too, I personally like the R2S very much, but it'll probably not make it to our test rigs and become 100% supported (e.g., RAUC, MAC addresses from a VPD prom, etc).  I'm also working on the Banana PI R3 router board (BPI-R3), which is the precursor for the R4 which seems to become the [OpenWRT reference board](https://forum.openwrt.org/t/openwrt-one-celebrating-20-years-of-openwrt/183684).

**Reply by wkz** [reply]
Microchip is definitely the future. Their stance on Open Source is great, actively developing and supporting open switchdev drivers.

I would just like to make a small amendment to @troglobit's message, because the codename-to-series mappings are not quite right. Just so there's no confusion on the Microchip families:

| Series | Codename | Description |
|-|-|-|
|VSC755x | Fireant | Up to 200G fabric with 25G SERDES |
| LAN969x | Laguna | Up to 102G fabric with 10G SERDES |
| LAN966x | Maserati | Small TSN switch with up to 8 ports of 1/2.5G |
| KSZxxxx | - | Low-cost switches |

**Reply by troglobit** [reply]
Updated list of currently supported boards are on our homepage https://kernelkit.org/platforms/
